### PR TITLE
Add glara keymaps: idobao, gmmk, kbdfans & mode

### PR DESCRIPTION
**Pull request by glara**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/iso/keymaps/glara/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+// #    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+// #    define DISABLE_RGB_MATRIX_SPLASH
+// #    define DISABLE_RGB_MATRIX_MULTISPLASH
+// #    define DISABLE_RGB_MATRIX_SOLID_SPLASH
+// #    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
```
Comment (waffle87) [inline keyboards/gmmk/pro/iso/keymaps/glara/config.h:70]:
```suggestion

#ifdef RGB_MATRIX_ENABLE
#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
#    define RGB_MATRIX_KEYPRESSES
#    define RGB_DISABLE_WHEN_USB_SUSPENDED
```

File: keyboards/gmmk/pro/iso/keymaps/glara/keymap.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    return true;
+}
+#endif
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/iso/keymaps/glara/keymap.c:103]:
```suggestion
```

File: keyboards/gmmk/pro/iso/keymaps/glara/rgb_matrix_map.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/iso/keymaps/glara/rgb_matrix_map.h:17]:
```suggestion
#pragma once

```

File: keyboards/gmmk/pro/iso/keymaps/glara/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = AW20216
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/iso/keymaps/glara/rules.mk:22]:
```suggestion
MOUSEKEY_ENABLE = no       # Mouse keys
```

File: keyboards/idobao/id80/v1/iso/keymaps/glara/rules.mk [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+VIA_ENABLE=yes
+
```
Comment (waffle87) [inline keyboards/idobao/id80/v1/iso/keymaps/glara/rules.mk:19]:
```suggestion
VIA_ENABLE = yes
```

File: keyboards/kbdfans/kbd75/keymaps/glara/rules.mk [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+LAYOUTS = 75_ansi 75_iso
+
```
Comment (waffle87) [inline keyboards/kbdfans/kbd75/keymaps/glara/rules.mk:21]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
* Your `gmmk/pro` keymap is in a non-existent location. It should be in `qmk_firmware/keyboards/gmmk/pro/rev1/iso/keymaps`.
* There is no such this as an `idobao/id80/v1/iso`, only `v2/iso`... Your keymap there should be moved accordingly or removed.
* You have an extra nested `glara` directory in your `kbdfans/kbd75/keymaps/glara` directory that needs to be removed...

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**